go/cmd/zk/zkcmd.go

```diff
@@ -133,35 +130,41 @@ func init() {
 	}
 }
 
-var (
-	server = flag.String("server", "", "server(s) to connect to")
-)
-
 func main() {
 	defer exit.Recover()
 	defer logutil.Flush()
+	server = pflag.String("server", "", "server(s) to connect to")
+
+	// handling case of --help, -help & -h
+	args := os.Args[1:]
+	var help bool
+	if args[0] == "-h" {
+		pflag.BoolVarP(&help, "help", "h", false, "display usage and exit")
+	} else if args[0] == "--help" || args[0] == "-help" {
+		pflag.BoolVar(&help, "help", false, "display usage and exit")
+	}
 
-	fs := pflag.NewFlagSet("zkcmd", pflag.ExitOnError)
-	log.RegisterFlags(fs)
-	logutil.RegisterFlags(fs)
-	_flag.SetUsage(flag.CommandLine, _flag.UsageOptions{ // TODO: hmmm
-		Epilogue: func(w io.Writer) { fmt.Fprint(w, doc) },
-	})
-	_flag.Parse(fs)
-	args := _flag.Args()
+	pflag.Parse()
+
+	if help || pflag.Arg(0) == "help" {
+		pflag.Usage()
+		os.Exit(0)
+	}
+
+	// if no zk command is provided after --server then we need to print doc & usage both
+	args = pflag.Args()
 	if len(args) == 0 {
-		flag.Usage()
+		fmt.Fprintf(os.Stderr, doc)
```

Comment:
you should override `pflag.CommandLine.Usage` instead of doing this

Reply:
done